src/hooks/useEventEmitter.ts

```diff
@@ -55,11 +86,19 @@ export function useEventEmitter(
             const eventListener = (...args) => savedHandler.current(...args);
 
             // Add event listener
-            emitter.on(eventName, eventListener);
+            if (isEventTarget(emitter) && typeof eventName === "string") {
+                emitter.addEventListener(eventName as string, eventListener);
```

Comment:
is the `as string` really needed here? I'd have hoped TS could deduce it from the typeof above